### ISSUE TITLE
Initialize signature showing Object.new signature but not the specific class signature

### ISSUE THREAD

**Issue by dgmora** [Shopify/ruby-lsp#2585]
### Reproduction steps

1. Create a class with an initialize method, give it no documentation.
2. In another place, instantiate this class
3. The signature help displays `class.rbs` help. It doesn't matter if the signature is different or no.
4. I would expect at least to see the signature.

This may be an issue with the neovim/lsp configuration, I'm wondering if other people see this too. I have seen this with my classes, but also with gems. The classes for the gems are loaded correctly, however when I try to instantiate it, I don't see the signature. Could be related to https://github.com/Shopify/ruby-lsp/pull/2553 ?

I've created a Dockerfile where you can see this happening. There are some annoying warnings, I need to update the Dockerfile.

```bash
curl -o Dockerfile https://gist.githubusercontent.com/dgmora/f869b0692db5fafdf6d43090ec9cb46f/raw/Dockerfile && \
  docker build -t lazyvim-lsp . && \
  docker run -it lazyvim-lsp
# `nvim` to start neovim
# inside:
#   `<space>`  to see available keymaps (and wait) 
#   `<space>e` to toggle filetree
#   `a`        to create a file in the file tree
```

![Image](https://github.com/user-attachments/assets/9e88b472-4274-41a8-b9dd-dce5de98379d)

**Comment by dgmora** [comment]
Thanks for the clarification!

I've also noticed that if the first class has a method like 
```ruby
def test
  puts 'hi'
end
```
Doing `MyExample.new.te` doesn't show `test`. Is that related?

**Comment by vinistock** [comment]
Not exactly. That would require some special handling for the `new` method in our type inferrer. 

However, we explored type flow analysis in #2421 and concluded that the effort vs benefit was not worth it at the moment (you essentially need to build an entire type checker to get that working). I say that because of this:

```ruby
# We can make this work properly because all we need to look into is the immediate receiver
MyExample.new.test

# This requires more sophisticated flow analysis, which we will not invest effort at this time
# Although this specific example might look simple, the mechanisms to ensure proper type flow analysis
# are not. We need to properly track scopes, create an internal type system so that we can represent
# unions if variables are set conditionally and many other fundamental pieces of a full fledged type
# checker
a = MyExample.new
a.test
```

**Comment by vinistock** [comment]
Reopening this as we didn't actually complete it, it was just added to the roadmap.
